webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultCsvValidationService.java

```diff
@@ -57,16 +58,20 @@ public DefaultCsvValidationService(final SchoolRepository schoolRepository) {
     }
 
     @Override
-    public List<CsvValidationFailure> validateHeaders(final Set<String> csvHeaders) {
+    public List<CsvValidationFailure> validateHeaders(final Map<String, Integer> csvHeaders) {
         final List<CsvValidationFailure> failures = new ArrayList<>();
-        final Set<String> headers = csvHeaders.stream().map(String::toLowerCase).collect(Collectors.toSet());
-        if (!headers.containsAll(RequiredHeaders)) {
-            final Set<String> missing = newHashSet(RequiredHeaders);
-            missing.removeAll(headers);
-            failures.add(CsvValidationFailure.builder()
-                    .row(0)
-                    .message("Missing headers " + missing)
-                    .build());
+
+        for (int i = 0; i < RequiredHeaders.size(); i++) {
+            final String header = RequiredHeaders.get(i);
+            if (!csvHeaders.containsKey(header) || csvHeaders.get(header) != i) {
+                failures.add(CsvValidationFailure.builder()
+                        .row(0)
+                        .message("Invalid headers. Headers must be in order: [" +
+                                Joiner.on(",").join(RequiredHeaders) +
+                                "]")
+                        .build());
+                break;
+            }
```

Comment:
The CSVParser gives us a Map<String, Integer> containing the header names and their index.  This just validates that the headers are in the expected order.

Reply:
Is there a ```parser.getHeaderList()``` ? Seems like it would be more straight forward to compare the order in which the headers appear with the order the required headers are in.
```if (!csvHeaders.get(i).equals(requiredHeaders.get(i))) { /* wrong order */ }```

Reply:
I wish.  But no, it was the first thing I looked for.

Reply:
Well that's just wacksauce